lib/matplotlib/rcsetup.py

```diff
@@ -299,6 +301,12 @@ def validate_color(s):
 
     raise ValueError('%s does not look like a color arg%s' % (s, msg))
 
+def deprecate_axes_colorcycle(value):
+    warnings.warn("axes.color_cycle is deprecated.  Use axes.prop_cycle "
+                  "instead. Will be removed in 2.1.0")
+    return validate_colorlist(value)
```

Comment:
This is going to have a very long deprecation cycle, at least a couple
years, given how long this has been in the api.

I could also be convinced that it should never be removed and keeping an
alias that sets the color cycle is a good idea.

On Sat, Aug 15, 2015, 3:01 PM OceanWolf <notifications@github.com> wrote:

> In lib/matplotlib/rcsetup.py
> <https://github.com/matplotlib/matplotlib/pull/4686#discussion_r37138302>:
>
> > @@ -299,6 +301,12 @@ def validate_color(s):
> >
> >      raise ValueError('%s does not look like a color arg%s' % (s, msg))
> >
> > +def deprecate_axes_colorcycle(value):
> > +    warnings.warn("axes.color_cycle is deprecated.  Use axes.prop_cycle "
> > +                  "instead. Will be removed in 2.1.0")
> > +    return validate_colorlist(value)
>
> I agree with the first point, but on the second you misunderstand me, I
> didn't mention setting prop_cycle in validate_colorlist, but in this
> deprecate_axes_colorcycle function. As by definition this function will
> get removed in 2.1.0, it makes sense to me to that here.
>
> I find it good practice to work around the deprecated stuff as early, aka
> as high up as possible in the codebase, rather then let it filter through
> to the deeper level code, as I see happens just above @tacaswell
> <https://github.com/tacaswell>'s comment "Should probably tag next
> version of cycler".
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4686/files#r37138302>.
>